File: taskflow-fullstack/frontend/src/services/project-service.ts
import { apiRequest } from '../lib/api-client'
import type { CreateProjectRequest, ProjectMemberResponse, ProjectResponse } from '../types/project'

export const projectService = {
  async getProjects(): Promise<ProjectResponse[]> {
    return apiRequest<ProjectResponse[]>('/api/v1/projects', {
      method: 'GET',
      authenticated: true,
    })
  },

  async getProjectById(projectId: number): Promise<ProjectResponse> {
    return apiRequest<ProjectResponse>(`/api/v1/projects/${projectId}`, {
      method: 'GET',
      authenticated: true,
    })
  },

  async getProjectMembers(projectId: number): Promise<ProjectMemberResponse[]> {
    return apiRequest<ProjectMemberResponse[]>(`/api/v1/projects/${projectId}/members`, {
      method: 'GET',
      authenticated: true,
    })
  },

  async createProject(payload: CreateProjectRequest): Promise<ProjectResponse> {
    return apiRequest<ProjectResponse>('/api/v1/projects', {
      method: 'POST',
      authenticated: true,
      body: JSON.stringify(payload),
    })
  },
}

File: taskflow-fullstack/backend/src/main/java/com/ruhatkaratas/taskflow/project/service/ProjectService.java
package com.ruhatkaratas.taskflow.project.service;

import com.ruhatkaratas.taskflow.project.dto.AddProjectMemberRequest;
import com.ruhatkaratas.taskflow.project.dto.CreateProjectRequest;
import com.ruhatkaratas.taskflow.project.dto.ProjectMemberResponse;
import com.ruhatkaratas.taskflow.project.dto.ProjectResponse;
import java.util.List;

public interface ProjectService {

    ProjectResponse createProject(CreateProjectRequest request, String currentUserEmail);

    List<ProjectResponse> getProjectsForUser(String currentUserEmail);

    ProjectResponse getProjectById(Long id, String currentUserEmail);

    List<ProjectMemberResponse> getProjectMembers(Long id, String currentUserEmail);

    ProjectMemberResponse addProjectMember(Long id, AddProjectMemberRequest request, String currentUserEmail);
}
